wow_message_parser/src/rust_printer/structs/print_common_impls/print_read.rs

```diff
@@ -835,6 +835,18 @@ pub(crate) fn print_read(s: &mut Writer, e: &Container, o: &Objects, prefix: &st
         return;
     }
 
+    // For fully compressed messages, replace the writer with a ZLibDecoder.
+    if e.tags().compressed() {
+        // Fully compressed messages always start with a u32 containing the decompressed size.
+        // We don't care about that, so we just ignore it.
```

Comment:
Can't `flate` pre allocate based on it?

Reply:
It seems as though there's a `new_with_buf` method that takes a buffer, but it doesn't allow for *just* preallocation. I think it still makes sense to do this, as `decompressed_size` is *probably* using `compressBound` from the C ZLib lib, which is guaranteed to be larger than the actual compressed thing?

Reply:
Done